### DISCUSSION TITLE
Traffic Race Mode

### DISCUSSION THREAD

**Discussion by sommo** [micbay/RaceTimerAndController#15] [General]
Hi @micbay,
what do you think about adding a Traffic Race Mode?
the lightstop (with adafruit bicolor bargraph), during the race, will work as a real lightstop and if you pass through while is red (you get a penalties, like -1 on your lap counter or it doesnt' count that lap ==> no +1 on your lap counter).
Do you like? is it possible?
I get my adafruit bicolor bargraph today, i will soldering and testing asap!
Thank you!

**Comment by micbay** [comment]
I think some game modes would be fun. I have as a project on the to-do list, developing a slot race system that is more game like, similar in concept to a live Mario Kart race where certain track segments with sensors will trigger randomly and the first car to cross it gets a bonus or a penalty, or an item like oil slick that they can release, via a car controller based button, that will cause all other lanes to have random power fluctuations. Stuff like that integrated with the track power so that different challenges that affect the drivers control and speed can be induced.

One of the initial goals for this particular embodiment was to see how cheap I could make it, which led to programming it to work, and fit on a Nano. I finished the code, and working on the documentation to release V2.0.0 for this controller today or tomorrow, and after adding the Drag Racing option, the program memory is 98% full. I have to delete most of the songs to get enough memory just to use the serial port for debugging. So to add much gamification will likely need to jump to a ATMega2560 based system with more program memory space.

You will be happy to know that I was able to integrate all of the features you asked for. You will be able to configure it to have a countdown on restarts from a pause, and you can set it up so that the songs are played based on finish order instead of racer name.

There is room for some game features, in trade for limiting audio to just 3 or 4 tunes, so I encourage open discussion on some gamification ideas and sensor/player trigger track integration build ideas.

**Comment by sommo** [comment]
Really happy to hear that!
the gamification is really amazing, i was thinking about slowdown cars if a car pass throught the stoplight while it's red for example but we need to control the power of every single line (that's why i propose to do a -1 lap from the racer's lap counter, maybe a faster solution but not so funny like the oil slick).
sure if i have some ideas i will share here!
thank you

**Comment by sommo** [comment]
it could be funny to have a training mode, just to test the circuit or to try to do a best lap, no matter for number of lap or who wins (no race mode). maybe a sound to tell you that you have done the best lap.

**Comment by sommo** [comment]
also a sound for the fastest lap during a normale race (to x lap or x lap on y time) could be awesome
